src/main/java/fr/insee/pearljam/batch/dao/CommunicationTemplateDaoImpl.java

```diff
@@ -17,12 +21,27 @@ public class CommunicationTemplateDaoImpl implements CommunicationTemplateDao {
     JdbcTemplate pilotageJdbcTemplate;
 
     @Override
-    public CommunicationTemplateType findById(Long id) {
-        String qString = "SELECT * FROM communication_template where communication_template.id= ?";
-        return pilotageJdbcTemplate.queryForObject(qString,
-                new CommunicationTemplateDaoImpl.CommunicationTemplateTypeMapper(), id);
+    public List<CommunicationTemplateType> findByMeshuggahIds(Set<String> meshuggahIds) {
+        StringBuilder qString = new StringBuilder("SELECT * FROM communication_template WHERE meshuggah_id IN (");
+
+        for (int i = 0; i < meshuggahIds.size(); i++) {
+            qString.append("?");
+            if (i < meshuggahIds.size() - 1) {
+                qString.append(", ");
+            }
+        }
+        qString.append(")");
+
+        return pilotageJdbcTemplate.query(
+            qString.toString(),
+            meshuggahIds.toArray(new String[0]),
+            new CommunicationTemplateDaoImpl.CommunicationTemplateTypeMapper()
+        );
```

Comment:
suggestion: could pre build the id concatenation then insert it in SQL statement, or could use jpa methodQuery to pass List<String> meshIds